### PR TITLE
chore(react-scheduler): add the Scheduler plugin API reference

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**

### CODE REVIEW

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+## User reference
+
+### Properties
+
+Name | Type | Default | Description
+-----|------|---------|------------
+data | Array&lt;object&gt; | | An array containing custom data. A user defines the access to this data.
+mapAppointmentData? | (appointment: object) => [Scheduler.MapAppointment](#schedulerrootprops) | | Specifies the map function used to get an appointment fields.
```
[the remark this answers is not in the document]
Comment (SergeyAlexeev) [reply packages/dx-react-scheduler/docs/reference/scheduler.md]: @viterobk check if the `AppointmentModel` is OK.

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { Scheduler } from '@devexpress/dx-react-scheduler';
+```
+
+## User reference
+
+### Properties
+
+Name | Type | Default | Description
+-----|------|---------|------------
+data | Array&lt;object&gt; | | An array containing custom data. A user defines the access to this data.
```
Comment (SergeyAlexeev) [inline packages/dx-react-scheduler/docs/reference/scheduler.md]: Why `custom data`?

Comment (MaximKudriavtsev) [reply packages/dx-react-scheduler/docs/reference/scheduler.md]: @SergeyAlexeev I changed the AppointmentModel Interface.

I suggest to define additonal properties by this [example](https://www.typescriptlang.org/docs/handbook/interfaces.html) 

```ts
interface SquareConfig {
    color?: string;
    width?: number;
    [propName: string]: any; // Additional properties
}
```

File: packages/dx-react-scheduler/docs/reference/scheduler.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (viterobk) [reply packages/dx-react-scheduler/docs/reference/scheduler.md]: OK

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Describes an appointment data object that should be returned by `mapAppointmentData` function.
+
+Field | Type | Description
+------|------|------------
+startDate | Date | The start date.
+endDate | Date | The end date.
+title? | string | The title.
+allDay? | boolean | The all day flag.
+id? | number &#124; string | The identifier.
+[propertyName: string] | any | Any number of other properties.
```
Comment (SergeyAlexeev) [inline packages/dx-react-scheduler/docs/reference/scheduler.md]: Does TS generator work with this string?

Comment (MaximKudriavtsev) [reply packages/dx-react-scheduler/docs/reference/scheduler.md]: Yes

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+## Interfaces
+
+### AppointmentModel
+
+Describes an appointment data object that should be returned by `mapAppointmentData` function.
+
+Field | Type | Description
+------|------|------------
+startDate | Date | The start date.
```
[the remark this answers is not in the document]
Comment (MaximKudriavtsev) [reply packages/dx-react-scheduler/docs/reference/scheduler.md]: I added the `string` type to startDate and endDate fields

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## Import
+
+Use the following statement to import a plugin with embedded theme components:
+
+```js
+import { Scheduler } from '@devexpress/dx-react-scheduler-material-ui';
+```
+
+If you want to use custom components, you can import the themeless plugin:
```
Comment (dirk-pieterse) [inline packages/dx-react-scheduler/docs/reference/scheduler.md]: Same change.

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+Name | Type | Default | Description
+-----|------|---------|------------
+data | Array&lt;[AppointmentModel](#appointmentmodel)&gt; | | An array of appointment data objects.
+rootComponent | ComponentType&lt;[Scheduler.RootProps](#schedulerrootprops)&gt; | | A component that renders the root layout.
+
+## Interfaces
+
+### AppointmentModel
+
+Describes an appointment data object that should be returned by `mapAppointmentData` function.
```
Comment (dirk-pieterse) [inline packages/dx-react-scheduler/docs/reference/scheduler.md]: Describes an appointment data object that should be returned by `mapAppointmentData` function -> Describes an appointment data object that the `mapAppointmentData` function should return.

File: packages/dx-react-scheduler/docs/reference/scheduler.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+Field | Type | Description
+------|------|------------
+startDate | Date &#124; string &#124; number | The start date.
+endDate | Date &#124; string &#124; number | The end date.
+title? | string | The title.
+allDay? | boolean | The all day flag.
+id? | number &#124; string | The identifier.
+[propertyName: string] | any | Any number of other properties.
```
Comment (dirk-pieterse) [inline packages/dx-react-scheduler/docs/reference/scheduler.md]: Any number of?

**Pull request merged**